.github/workflows/lint.yml

```diff
@@ -63,11 +64,16 @@ jobs:
           tox -e ruff -- check
 
       - name: Run linting
-        if: success() || failure()
+        if: ${{ !cancelled() && (steps.deps.outcome == 'success') }}
         run: |
           echo "::add-matcher::.github/workflows/matchers/pylint.json"
           tox -e lint
 
+      - name: Run mypy type checks
+        if: ${{ !cancelled() && (steps.deps.outcome == 'success') }}
+        run: |
+          tox -e mypy
```

Comment:
We can also add a matcher like is done above for lint.

I found a matcher we can copy: 
https://github.com/home-assistant/core/blob/dev/.github/workflows/matchers/mypy.json

Reply:
We can...